File: Big product/Big product.go
// https://basecamp.eolymp.com/en/problems/9001

package main

import (
	"fmt"
	"math"
)

func main() {
	var a, b int64
	fmt.Scan(&a, &b)

	if a > 0 && b > 0 {
		fmt.Println(1)
	} else if (a < 0 && b > 0) || a == 0 || b == 0 {
		fmt.Println(0)
	} else if a < 0 && b < 0 {
		if (int64(math.Abs(float64(b-a)))+1)%2 == 1 {
			fmt.Println(-1)
		} else {
			fmt.Println(1)
		}
	}
}

File: Big product/Big product.java
// https://basecamp.eolymp.com/en/problems/9001

import java.util.Scanner;

public class BigProduct {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long a = sc.nextLong();
        long b = sc.nextLong();

        if (a > 0 && b > 0) {
            System.out.print(1);
        } else if ((a < 0 && b > 0) || a == 0 || b == 0) {
            System.out.print(0);
        } else if (a < 0 && b < 0) {
            long diff = Math.abs(b - a) + 1;
            if (diff % 2 == 1) {
                System.out.print(-1);
            } else {
                System.out.print(1);
            }
        }
    }
}
